### PR TITLE
Build Updates:

### PR DESCRIPTION
**Pull request by seando-adsk**
* For VisualStudio make some warnings (unused variable) as errors.
* Don't need gmock.
* Added vscode files to ignore.

### CODE REVIEW

File: cmake/compiler_config.cmake [hunk 1]
```diff
@@ -53,6 +53,9 @@ set(MSVC_FLAGS
     /wd4180
     # exporting STL classes
     /wd4251
+    # Set some warnings as errors (to make it similar to Linux)
+    /we4101
```
Comment (seando-adsk) [inline cmake/compiler_config.cmake:57]: 'identifier' : unreferenced local variable

File: cmake/compiler_config.cmake [hunk 2]
```diff
@@ -53,6 +53,9 @@ set(MSVC_FLAGS
...
+    /we4189
```
Comment (seando-adsk) [inline cmake/compiler_config.cmake:58]: 'identifier' : local variable is initialized but not referenced

File: cmake/googletest_download.txt.in [hunk 3]
```diff
@@ -25,6 +25,7 @@ ExternalProject_Add(googletest
                     "-DCMAKE_INSTALL_PREFIX=${GOOGLETEST_BUILD_ROOT}/googletest-install"
                     "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
                     "-Dgtest_force_shared_crt=ON"
+                    "-DBUILD_GMOCK=OFF"
```
Comment (seando-adsk) [inline cmake/googletest_download.txt.in:28]: We don't use gmock so no point in building it.

**Review by JGamache-autodesk — APPROVED** [review]

**Comment by seando-adsk** [comment]
I ran the preflight manually (as our internal build system is currently not working because of a GitHub API change) and it succeeded.

**Pull request merged**